File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/atts/proc/AttPrefixSuffixProcessor.java
package ru.citeck.ecos.records3.record.op.atts.proc;

import lombok.Data;
import lombok.RequiredArgsConstructor;
import org.jetbrains.annotations.NotNull;
import ru.citeck.ecos.commons.data.DataValue;
import ru.citeck.ecos.commons.data.ObjectData;

import java.util.List;

public class AttPrefixSuffixProcessor extends AbstractAttProcessor<AttPrefixSuffixProcessor.Args> {

    @Override
    protected Object processOne(@NotNull ObjectData meta, @NotNull DataValue value, @NotNull Args args) {
        return args.prefix + value.asText() + args.suffix;
    }

    @Override
    protected Args parseArgs(@NotNull List<DataValue> arguments) {

        String prefix = "";
        String suffix = "";

        if (arguments.size() > 0) {
            prefix = arguments.get(0).asText();
        }
        if (arguments.size() > 1) {
            suffix = arguments.get(1).asText();
        }
        return new Args(prefix, suffix);
    }

    @NotNull
    @Override
    public String getType() {
        return "presuf";
    }

    @Data
    @RequiredArgsConstructor
    public static class Args {
        private final String prefix;
        private final String suffix;
    }
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/atts/service/proc/AttPrefixSuffixProcessor.kt
package ru.citeck.ecos.records3.record.op.atts.service.proc

import ru.citeck.ecos.commons.data.DataValue
import ru.citeck.ecos.commons.data.ObjectData

class AttPrefixSuffixProcessor : AbstractAttProcessor<AttPrefixSuffixProcessor.Args>() {

    override fun processOne(attributes: ObjectData, value: DataValue, args: Args): Any? {
        return args.prefix + value.asText() + args.suffix
    }

    override fun parseArgs(args: List<DataValue>): Args {
        var prefix = ""
        var suffix = ""
        if (args.isNotEmpty()) {
            prefix = args[0].asText()
        }
        if (args.size > 1) {
            suffix = args[1].asText()
        }
        return Args(prefix, suffix)
    }

    override fun getType(): String = "presuf"

    class Args(val prefix: String, val suffix: String)
}
